Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -6319,12 +6319,36 @@ void WebPageProxy::didFailProvisionalLoadForFrameShared(Ref<WebProcessProxy>&& p
     m_failingProvisionalLoadURL = provisionalURL;
 
     if (willInternallyHandleFailure == WillInternallyHandleFailure::No) {
-        if (m_loaderClient)
-            m_loaderClient->didFailProvisionalLoadWithErrorForFrame(*this, frame, navigation.get(), error, process->transformHandlesToObjects(userData.protectedObject().get()).get());
-        else {
-            m_navigationClient->didFailProvisionalNavigationWithError(*this, FrameInfoData { frameInfo }, navigation.get(), request.url(), error, process->transformHandlesToObjects(userData.protectedObject().get()).get());
-            m_navigationClient->didFailProvisionalLoadWithErrorForFrame(*this, WTFMove(request), error, WTFMove(frameInfo));
-        }
+        auto callClientFunctions = [this, frame = Ref { frame }, navigation, error, process = process, request = WTFMove(request), frameInfo = WTFMove(frameInfo), protectedObject = userData.protectedObject()]() mutable {
```

Comment:
nit: I don't think you need the `= process` here.

Reply:
Also, the lambda seems to fail to protect `this`. I think we need to capture a `protectedThis`

Reply:
> the lambda seems to fail to protect this. I think we need to capture a protectedThis

The lambda is either called synchronously, or we protect `this` in the other completion handler we pass into `protectedPageClient->showBrowsingWarning` (below), but I'll protect `this` in this lambda, so it is clear to anyone else looking at it.